pkg/util/maven/maven_project_test.go

```diff
@@ -181,6 +181,17 @@ func TestParseGAVMvnNoVersion(t *testing.T) {
 	assert.Equal(t, dep.ArtifactID, "camel-core")
 }
 
+func TestParseGAVMvnClassifierNoVersion(t *testing.T) {
+	dep, err := ParseGAV("org.apache.camel:camel-k-core:jar:custom")
```

Comment:
I'd do a permutation of the possible cases to make sure we're not missing something. We should have at least 7 different cases to test.

Reply:
The only missing test case is group:artifact:classifier (no type, no version), as the current regex grouping does some assumption, which don't support the mentioned test case and I think it's not worth to spend much time to change this regex to accommodate this particular use case.
We can add a requirement, if there is a classifier, then a type is also required.